egui/src/widgets/plot/mod.rs

```diff
@@ -182,6 +187,35 @@ impl Plot {
         self
     }
 
+    /// Provide a function to customize the on-hovel label for the x and y axis
+    ///
+    /// ```
+    /// # egui::__run_test_ui(|ui| {
+    /// use egui::plot::{Line, Plot, Value, Values};
+    /// let sin = (0..1000).map(|i| {
+    ///     let x = i as f64 * 0.01;
+    ///     Value::new(x, x.sin())
+    /// });
+    /// let line = Line::new(Values::from_values_iter(sin));
+    /// Plot::new("my_plot").view_aspect(2.0)
+    /// .custom_label_func(|name, value| {
+    ///     if let Some(name) = name {
+    ///         format!("{}: {:.*}%", name, 1, value.y).to_string()
+    ///     } else {
+    ///         "".to_string()
+    ///     }
+    /// })
+    /// .show(ui, |plot_ui| plot_ui.line(line));
+    /// # });
+    /// ```
+    pub fn custom_label_func<F: 'static + Fn(Option<&str>, &Value) -> String>(
```

Comment:
I think it would be simpler to use `&str` instead of `Option<&str>`. An empty string is a good way to communicate "there is no name for this". The type `Option<&str>` tells users "this could be `Some("")`" which means they should guard against that, which is annoying.

Reply:
The `'static` bound may be a bit annoying for some users, but adding a lifetime bond to `Plot` would also be annoying, so perhaps this is a good enough start.

Reply:
> I think it would be simpler to use `&str` instead of `Option<&str>`. An empty string is a good way to communicate "there is no name for this". The type `Option<&str>` tells users "this could be `Some("")`" which means they should guard against that, which is annoying.

No problem. I have reverted the commit so that we switch back to `&str`.